atomic_reactor/utils/resource_manager.py

```diff
@@ -0,0 +1,132 @@
+"""
+Copyright (c) 2022 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+import os
+import subprocess
+import logging
+import json
+from datetime import datetime
+from shlex import quote
+
+ROOT_PATH = 'resource_manager/'
+logger = logging.getLogger(__name__)
+
+
+class ResourceManager:
+    '''
+        Manages build resources
+    '''
+
+    def __init__(self):
+        pass
+
+    def check_avail(self, user, host):
+        try:
+            data = os.popen(f'ssh {user}@{host} -4C "cat {ROOT_PATH}info.json"').read()
+            if data == '':
+                logger.warning("%s hasn't been deployed yet", host)
+                return False
+            else:
+                # parse slot count, arch
+                meta = json.loads(data)
+                logger.debug("%s is up", meta['hostname'])
+
+                # check slots status
+                meta['slots'] = []
+                free_slots = 0
+                i = 0
+                while i < meta.get('max_slot_count', 1):
```

Comment:
Why not `for` cycle when we know number of iterations in advance?

Reply:
Do you insist here on using while instead of for @tveverka-RH ? Code would be more readable with `for` and you would avoid incrementing index.

Reply:
May I ask how do you want to go with for, unless you will list all slots from the directory? 
In advance I know only slot count. 

I do not insist on while statement, just don't know how do you want to proceed with for

Reply:
```suggestion
                for i in range(meta.get('max_slot_count', 1)):
```
or am I missing something?

Reply:
Ok, dumb me. Fixed in the last commit